packages/headless/src/api/analytics/analytics-crypto-polyfill.ts

```diff
@@ -0,0 +1,19 @@
+interface CryptoModuleInNode {
+  getRandomValues?: typeof crypto.getRandomValues;
+  webcrypto?: {getRandomValues: typeof crypto.getRandomValues};
+}
+
+const getGlobalCrypto = () => global.crypto as CryptoModuleInNode;
+
+export const polyfill = () => {
```

Comment:
adding a TODO / doc about when this can be removed in future would be useful

Reply:
Naming it specifically would be helpful too even though this is the only polyfill e.g `polyfillCryptoNode17`

Reply:
We could: https://nodejs.org/api/globals.html#crypto_1
Starting with Node 20, crypto as a global property is enabled by default.
Assuming we keep on supporting 2 majors version of Node at a time, when Node 22 will roll out in ~Y24Q4, we could remove the polyfill.